### PR TITLE
SidebySide Layout

### PR DESCRIPTION
**Pull request by ThaMighty90**
This is an easy to use SidebySide Layout screens match perfectly in size and it is much easier then tinkering in the ini file

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/2859)

### CODE REVIEW

**Comment by ThaMighty90** [comment]
Yeah sry for that didn't watch carefully enough. VS just thought I changed something in them

**Comment by wwylele** [comment]
fyi the files in `x64` is generated by VS and due to [a bug in cmake](https://gitlab.kitware.com/cmake/cmake/issues/16458) they are not in the hidden `build` directory, so you need to manually ignore/unstage them when committing in git

File: src/core/frontend/framebuffer_layout.cpp [hunk 1]
```diff
@@ -109,11 +109,10 @@ FramebufferLayout LargeFrameLayout(unsigned width, unsigned height, bool swapped
     // To do that, find the total emulation box and maximize that based on window size
     float window_aspect_ratio = static_cast<float>(height) / width;
     float emulation_aspect_ratio =
-        swapped
```
Comment (ThaMighty90) [inline src/core/frontend/framebuffer_layout.cpp]: Unintended!

File: src/core/frontend/framebuffer_layout.cpp [hunk 2]
```diff
@@ -141,6 +140,48 @@ FramebufferLayout LargeFrameLayout(unsigned width, unsigned height, bool swapped
...
+        maxRectangle(screen_window_area, BOT_SCREEN_ASPECT_RATIO);
+
+    float window_aspect_ratio = static_cast<float>(height) / width;
+    // both screens width are taken into account by multiplying by 2
+    float emulation_aspect_ratio = TOP_SCREEN_ASPECT_RATIO * 2;
+
+    if (window_aspect_ratio < emulation_aspect_ratio) {
+
+        // Apply borders to the left and right sides of the window.
+        top_screen = top_screen.TranslateX(screen_window_area.GetWidth() -
```
Comment (ThaMighty90) [inline src/core/frontend/framebuffer_layout.cpp]: Changing it as some I get on the PC weekend is family time :)

**Comment by ThaMighty90** [comment]
leftrightbars
topbottombars

**Comment by jroweboy** [comment]
Nice job on the update. Heres a short todo list of things left to fix

* Need to remove those extra files
* Need to add an option for your layout to the qt settings dialog
* Need to fix the style issue here https://travis-ci.org/citra-emu/citra/jobs/261625189#L1208

After we get those updated, i'll look over the changes again.

**Comment by jroweboy** [comment]
nope. not how to do that. all you have to do is edit https://github.com/citra-emu/citra/blob/1631e99eed644bb9d313bc49ef6023ac3afe1e52/src/citra_qt/configuration/configure_graphics.ui#L146 this xml file and add the new option. please remove the ui.h file that you added. that file is autogenerated from the xml file that i linked you (its breaking the build)

File: src/core/frontend/framebuffer_layout.cpp [hunk 3]
```diff
@@ -141,6 +141,43 @@ FramebufferLayout LargeFrameLayout(unsigned width, unsigned height, bool swapped
...
+    MathUtil::Rectangle<unsigned> screen_rect =
+        maxRectangle(screen_window_area, SIDE_BY_SIDE_ASPECT_RATIO);
+    // Find sizes of top and bottom screen
+    MathUtil::Rectangle<unsigned> top_screen = maxRectangle(screen_rect, TOP_SCREEN_ASPECT_RATIO);
+    MathUtil::Rectangle<unsigned> bot_screen = maxRectangle(screen_rect, BOT_SCREEN_ASPECT_RATIO);
+
+    if (window_aspect_ratio < emulation_aspect_ratio) {
+        // Apply borders to the left and right sides of the window.
+        top_screen =
+            top_screen.TranslateX((screen_window_area.GetWidth() - screen_rect.GetWidth()) / 2);
```
Comment (jroweboy) [inline src/core/frontend/framebuffer_layout.cpp]: pull the translate amount into a variable `u32 shift = ...` for example (and do this for both if and else)

File: src/core/frontend/framebuffer_layout.cpp [hunk 4]
```diff
@@ -141,6 +141,43 @@ FramebufferLayout LargeFrameLayout(unsigned width, unsigned height, bool swapped
...
 
+FramebufferLayout SideFrameLayout(unsigned width, unsigned height, bool swapped) {
+    ASSERT(width > 0);
+    ASSERT(height > 0);
+
+    FramebufferLayout res{width, height, true, true, {}, {}};
+    // Aspect ratio of both screens side by side
+    const float SIDE_BY_SIDE_ASPECT_RATIO = static_cast<float>(Core::kScreenTopHeight) /
+                                            (Core::kScreenTopWidth + Core::kScreenBottomWidth);
+    float emulation_aspect_ratio = SIDE_BY_SIDE_ASPECT_RATIO;
```
Comment (jroweboy) [inline src/core/frontend/framebuffer_layout.cpp]: useless variable

File: src/citra_qt/configuration/configure_graphics.ui [hunk 5]
```diff
@@ -159,6 +159,11 @@
               <string notr="true">Large Screen</string>
              </property>
             </item>
+             <item>
+               <property name="text">
+                 <string notr="true">Side by Side</string>
```
Comment (jroweboy) [inline src/citra_qt/configuration/configure_graphics.ui]: can we make this translatable? (I'm not sure why they are `notr` right now but its probably fine to translate) also change the other options in this dropdown to be translatable too

Comment (ThaMighty90) [inline src/citra_qt/configuration/configure_graphics.ui]: so how particular do i go about that? i never worked with qt...do i just get rid of the notr="true"?

Comment (jroweboy) [inline src/citra_qt/configuration/configure_graphics.ui]: yup thats all it takes is removing the notr properties

for my last few nit picks (sorry!) 

* `shift_horizontal` and `shift_vertical` are only used in the respective branches, so they should be defined in those branches (`shift_horizontal` in the if and `shift_vertical` in the else)
* both of those variables should be u32 (int is platform defined size)

File: src/citra_qt/configuration/configure_graphics.ui [hunk 6]
```diff
@@ -146,19 +146,24 @@
...
              </property>
             </item>
             <item>
              <property name="text">
-              <string notr="true">Large Screen</string>
+              <string>Large Screen</string>
              </property>
             </item>
+             <item>
+               <property name="text">
```
Comment (wwylele) [inline src/citra_qt/configuration/configure_graphics.ui]: nit: indention is off for both `<item>` and `<property>`

File: src/core/frontend/framebuffer_layout.h [hunk 7]
```diff
@@ -54,6 +54,17 @@ FramebufferLayout SingleFrameLayout(unsigned width, unsigned height, bool is_swa
 FramebufferLayout LargeFrameLayout(unsigned width, unsigned height, bool is_swapped);
 
 /**
+* Factory method for constructing a Frame with the a 1x size Top screen with a 1x size bottom
+* screen side by side
+* This is useful for devices like the GPDWin
+* @param width Window framebuffer width in pixels
+* @param height Window framebuffer height in pixels
+* @param is_swapped if true, the bottom screen will be the large display
```
Comment (wwylele) [inline src/core/frontend/framebuffer_layout.h]: `the bottom screen will be the large display` -> `the bottom screen will be displayed on the left`?

**Pull request merged**